gnat2goto/driver/tree_walk.adb

```diff
@@ -765,13 +770,21 @@ package body Tree_Walk is
               Store_Nat_Bound (Bound_Type_Nat (Intval (Upper_Bound)));
          when N_Attribute_Reference => Upper_Bound_Value :=
               Store_Symbol_Bound (Get_Array_Attr_Bound_Symbol (Upper_Bound));
+         when N_Identifier =>
+            Upper_Bound_Value :=
+               Store_Symbol_Bound (Bound_Type_Symbol (Upper_Bound));
          when others =>
             Report_Unhandled_Node_Empty (Upper_Bound,
                                          "Do_Base_Range_Constraint",
                                          "unsupported upper range kind");
       end case;
 
-      Set_Width (Result_Type, Get_Width (Resolved_Underlying));
+      if Kind (Resolved_Underlying) = I_C_Enum_Type then
```

Comment:
This seems sketchy. `enumeration_typet` does not have a width field, seems like you're basically invoking UB here?

Reply:
The `enumeration_typet` doesn't have a `width` field, but it has a subtype that has a `width` field.